components/scream/src/share/io/scream_output_manager.cpp

```diff
@@ -90,7 +90,7 @@ setup (const ekat::Comm& io_comm, const ekat::ParameterList& params,
     }
   }
 
-  const auto has_restart_data = (m_avg_type!=OutputAvgType::Instant || m_output_control.frequency>1);
+  const auto has_restart_data = (m_avg_type!=OutputAvgType::Instant && m_output_control.frequency>1);
```

Comment:
has_restart_data is used for both Checkpoint_control and history output_control. Not sure if the original logic is necessary for Checkpoint. It definitely needs to be changed for history output control. And for history output restart, the 2nd condition is not necessary, and  inappropriate if "Frequency Units" is not 'Steps".

Reply:
I don't really know what any of these variables mean, but I conjecture that the intent was that we should require the existence of a history restart file if output isn't instantaneous and if we're not restarting at the start of an averaging interval for output.